client/post-editor/editor-visibility/index.jsx

```diff
@@ -65,10 +69,23 @@ export default React.createClass( {
 		}
 	},
 
+	getVisibility( props ) {
+		if ( props.password ) {
+			return 'password';
+		}
+
+		if ( 'private' === props.status ) {
+			return 'private';
+		}
+
+		return 'public';
+	},
+
 	setVisibility( props ) {
-		if ( props.visibility !== this.state.visibility ) {
+		const newVisibility = this.getVisibility( props );
```

Comment:
At this point I question whether `visibility` should exist in state at all, and instead to simply call upon `getVisibility` directly at any point we'd have used the state value.